server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/GlobalQuotaRoutes.java

```diff
@@ -55,42 +71,95 @@ public GlobalQuotaRoutes(MaxQuotaManager maxQuotaManager, JsonTransformer jsonTr
+    })
+    public void updateQuota() {
+        service.put(QUOTA_ENDPOINT, ((request, response) -> {
             try {
-                QuotaRequest quotaRequest = QuotaRequest.parse(request.body());
-                maxQuotaManager.setDefaultMaxMessage(quotaRequest.getValue());
+                QuotaDTO quotaDTO = jsonExtractor.parse(request.body());
+                maxQuotaManager.setDefaultMaxMessage(quotaDTO.getCount());
+                maxQuotaManager.setDefaultMaxStorage(quotaDTO.getSize());
                 response.status(204);
+            } catch (JsonExtractException e) {
+                LOGGER.info("Malformed JSON", e);
+                response.status(400);
             } catch (IllegalArgumentException e) {
-                LOGGER.info("Invalid quota. Need to be an integer value greater than 0");
+                LOGGER.info("Quota should be positive or unlimited (-1)", e);
                 response.status(400);
             }
             return Constants.EMPTY_BODY;
-        });
+        }));
+    }
 
-        service.get(SIZE_ENDPOINT, (request, response) -> {
-            long value = maxQuotaManager.getDefaultMaxStorage();
+    @GET
+    @ApiOperation(value = "Reading count and size at the same time")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK", response = QuotaDTO.class),
+            @ApiResponse(code = 500, message = "Internal server error - Something went bad on the server side.")
+    })
+    public void getQuota() {
+        service.get(QUOTA_ENDPOINT, (request, response) -> {
+            QuotaDTO quotaDTO = QuotaDTO.builder()
+                .count(maxQuotaManager.getDefaultMaxMessage())
+                .size(maxQuotaManager.getDefaultMaxStorage()).build();
             response.status(200);
-            return value;
+            return quotaDTO;
         }, jsonTransformer);
+    }
 
+    @DELETE
+    @Path("/size")
+    @ApiOperation(value = "Removing per quotaroot mail size limitation by updating to unlimited value")
+    @ApiResponses(value = {
+            @ApiResponse(code = 204, message = "The value is updated to unlimited value."),
+            @ApiResponse(code = 500, message = "Internal server error - Something went bad on the server side.")
+    })
+    public void deleteQuotaSize() {
         service.delete(SIZE_ENDPOINT, (request, response) -> {
             maxQuotaManager.setDefaultMaxStorage(Quota.UNLIMITED);
             response.status(204);
             return Constants.EMPTY_BODY;
         });
+    }
 
+    @PUT
+    @Path("/size")
+    @ApiOperation(value = "Updating per quotaroot mail size limitation")
+    @ApiImplicitParams({
+            @ApiImplicitParam(required = true, dataType = "integer", paramType = "body")
+    })
+    @ApiResponses(value = {
+            @ApiResponse(code = 204, message = "OK. The value updated."),
+            @ApiResponse(code = 400, message = "The body is not a positive integer."),
```

Comment:
Checked code. Unlimited value is just accept in case of update count and size in the same time.
In case of update separated count or size, value must be > 0.